src/app/AutoResubscribeClient.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/AutoResubscribeClient.h>
+#include <app/InteractionModelEngine.h>
+
+namespace chip {
+namespace app {
+
+CHIP_ERROR AutoResubscribeClient::Init(Messaging::ExchangeManager * apExchangeMgr, ReadClient::Callback * apCallback,
+                                       ResubscribePolicyDelegate * apResubscribeDelegate)
+{
+    mpExchangeMgr = apExchangeMgr;
+    mCallback.Init(apCallback, this);
+    if (apResubscribeDelegate == nullptr)
+    {
+        mpResubscribePolicyDelegate = &mDefaultResubscribeImpl;
+    }
+    else
+    {
+        mpResubscribePolicyDelegate = apResubscribeDelegate;
+    }
+    app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
+    return engine->NewReadClient(&mpReadClient, ReadClient::InteractionType::Subscribe, &mCallback);
```

Comment:
This approach will not be compatible with #13108 which gives the flexibility to the application to figure out how best to allocate the ReadClient object.

Reply:
Per offline chat with @bzbarsky-apple @mrjerryjohns , decides to put resubscribe logic inside readClient. update the PR, thanks